### PR TITLE
Fixes medisim combatants falling to the lower z-level on Icebox

### LINKED ISSUE #56695: [BUG] Icebox has unintended interactions with Medieval Simulation Shuttle
## Testmerges: Inapplicable

image

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

This adds a subtype of the chasm turf, which skips over attempting any z-level stuff and skips to the Part Where It Kills You.

This replaces the chasms in the medisim shuttle, because knights kept falling down the chasms and escaping containment.
## Why It's Good For The Game

Closes #56695.
## Changelog
:cl: Rhials
fix: Falling down one of the stage hazard chasms on the medisim shuttle will now properly kill you on Icebox, instead of dropping you into the z-level below.
/:cl:

### CODE REVIEW

File: code/datums/components/chasm.dm [hunk 1]
```diff
@@ -208,6 +210,10 @@ GLOBAL_LIST_INIT(chasm_storage, list())
 	SIGNAL_HANDLER
 	UnregisterSignal(gone, COMSIG_LIVING_REVIVE)
 
+///A type of chasm that bypasses the check to drop objects down a z-level, throwing whatever falls straight into the chasm contents.
+/datum/component/chasm/offstage
+	respect_zlevel = FALSE
```
Comment (san7890) [inline code/datums/components/chasm.dm:215]: do you think this would benefit if we made this an argument on initialize instead of a subtype or

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/datums/components/chasm.dm (hunk 1 above)
Comment (Fikou) [reply code/datums/components/chasm.dm:215]: yeah you already override addcomponent

**Review by Fikou — CHANGES_REQUESTED** [review]
What's the point of the new component subtype? You can just not supply the addcomponent with the target turf arg to make it not drop people down.

File: code/datums/components/chasm.dm (hunk 1 above)
Comment (tralezab) [reply code/datums/components/chasm.dm:215]: I would just make this an arg, default to TRUE to make admin's lives easier

**Pull request merged**